src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets

```diff
@@ -149,13 +149,15 @@ Copyright (c) .NET Foundation. All rights reserved.
     but were referenced transitively via a project or package reference.  NuGet writes these
     to the assets file, and the ResolvePackageAssets target adds them to the TransitiveFrameworkReference
     item.  Here, we add them to FrameworkReference if they aren't already referenced.
+    We add a transitive attribute to the FrameworkReferences so NuGet can filter them out from CollectFrameworkReferences.
     ============================================================
     -->
   <Target Name="AddTransitiveFrameworkReferences" AfterTargets="ResolvePackageAssets"
           Condition="'@(TransitiveFrameworkReference)' != ''" >
 
     <ItemGroup>
-      <FrameworkReference Include="@(TransitiveFrameworkReference)" Exclude="@(FrameworkReference)"/>
+      <FrameworkReference Include="@(TransitiveFrameworkReference)" Exclude="@(FrameworkReference)"
+                          IsTransitiveFrameworkReference="true" />
```

Comment:
Can we set `DisableTransitiveFrameworkReferences` property to `false` if we are not interested in `TransitiveFrameworkReference` at all so that we can avoid `NuGet` changes. I am not sure of the side effects though :)

https://github.com/dotnet/sdk/blob/67d902bf9382641e63f993da916784c6a268b553/src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs#L91

Reply:
@kartheekp-ms I think that can have a correctness problem. 

Props declared in targets will `leak` into the whole target execution.